alert/expiring.py

```diff
@@ -87,10 +95,17 @@ def get_release_dates():
     
     # scrape for past release date tables
     table = soup.find(id="Past_branch_dates").find_parent("h2").find_next_sibling("table")
-    for i, row in enumerate(table.find_all("tr")):
-        if i == 0: continue # skip the header row
+    for row in table.find_all("tr"):
+        # obtain the row and make sure it is valid (this skips the header row and any minor version rows)
         fields = list(row.find_all("td"))
-        if len(fields) != 4: continue # row with some blank entries
+        if len(fields) < 4: continue # not enough fields in the row, probably a header row
+        is_valid = True
+        for field in fields[-4:]: # ensure that each column represents a Firefox version
+            if "Firefox" not in field.string:
+                is_valid = False
+                break
+        if not is_valid: continue
+
         nightly_version = str(version_get_major(fields[-4].string.replace("Firefox ", ""))) + ".0a1"
```

Comment:
Scraping for past and future release date tables should be refactored into a single function.